provider/oci/environ.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2018 Canonical Ltd.
```

Comment:
I'm skipping the next few files because I think it is all included in the 1/7 section.
I'm not sure what's going to be a better way for us to review chained changes like this.

In things-that-aren't-github I've used pull requests that reference another pull request and then they can show just the new diff.

One option is to have you present these as a series of branches that are to be merged, and then propose each merge into the previous.

eg:

oci-1
oci-2
oci-3

pr #1 = oci-1=>juju/develop
pr #2 = oci-2 => oci-1
pr #3 = oci-3 => oci-2

Reply:
That is actually how I created the branches. Unfortunately, github is lacking here. We can't really do chained PRs, in a way that will only show the diff against the previous PR. The only sensible way to get small diffs in chained branches is if all branches are in the same repository, and we set the parent branch to the previously proposed branch.

So we are stuck with this for now. As we merge the first branches, the diffs will get smaller on the later branches... Really sorry about this.